torch/backends/linalg/__init__.py

```diff
@@ -0,0 +1,43 @@
+import sys
+import torch
+from contextlib import contextmanager
+from torch.backends import ContextProp, PropModule, __allow_nonbracketed_mutation
+
+def set_flags(_preferred=None):
+    orig_flags = (torch._C._get_linalg_preferred_backend(),)
+    if _preferred is not None:
+        if not isinstance(_preferred, torch.linalg_backend):
+            raise RuntimeError("must choose a linalg backend from: "
+                               "torch.linalg_default, torch.linalg_cusolver, torch.linalg_magma.")
+        torch._C._set_linalg_preferred_backend(_preferred)
+    return orig_flags
+
+
+@contextmanager
```

Comment:
@ngimel points out that because this setting is global we shouldn't provide a context manager for it -- the context manager wouldn't be respected when multithreading